### PR TITLE
Run a flush cycle before calling model promise callbacks.

### PR DESCRIPTION
**Pull request by burrows**
This ensures that any cached props that are invalidated by a mapper response
will be updated before the promise callback is called.

```javascript
const Transis = require('../dist');

const PersonMapper = {
  get: function(id) {
    return new Promise(function(resolve, reject) {
      setTimeout(function() {
        resolve({id: id, firstName: 'Homer', lastName: 'Simpson'});
      }, 5);
    });
  }
};

const Person = Transis.Model.extend('Person', function() {
  this.mapper = PersonMapper;

  this.attr('firstName', 'string');
  this.attr('lastName', 'string');

  this.prop('fullName', {
    cache: true,
    on: ['firstName', 'lastName'],
    get: function(firstName, lastName) {
      return firstName + ' ' + lastName;
    }
  });
});

let p = Person.get(1);
console.log('get:', p.fullName);

p.then(function() {
  console.log('then callback:', p.fullName);
});
```
 Previously this logged:

```
get: undefined undefined
then callback: undefined undefined
```

But now it logs:

```
get: undefined undefined
then callback: Homer Simpson
```

### CODE REVIEW

File: spec/helpers.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import "es6-shim";
 import * as util from "../util";
 import TransisObject from "../object";
 
-function delay(f) { setTimeout(f, 5); }
+function delay(f) { setTimeout(f, 10); }
```
Comment (burrows) [inline spec/helpers.js]: This was need to ensure that specs pass reliably on phantomjs.

**Comment by burrows** [comment]
Note to reviewers: I committed the compiled files in the `dist/` directory on this branch in order to allow for installing the module from a github url. Without those files committed you get a broken install otherwise. This allows us to test Transis changes in CI without having to release a new npm module.

**Comment by alexraginskiy** [comment]
@burrows is there a reason there are so many files in the dist?  I believe the package consumes index.js, not the individual source files.

**Comment by burrows** [comment]
@alexraginskiy The index.js requires the other files.  Are you thinking of the webpack package? I just added that as well, but we don't actually use that file.

**Pull request merged**